File: src/sokol_impl.m
// *************************************************************************
// *                      RevenantRevisited (port)                         *
// *  sokol_impl.m - Sokol implementation TU for Apple / Metal builds.     *
// *                                                                       *
// *  macOS only: sokol_app + sokol_gfx share the Metal backend and need   *
// *  to be compiled as Objective-C. On Windows / Linux, CMake drops this  *
// *  file and compiles sokol_impl.c against SOKOL_D3D11 / SOKOL_GLCORE33  *
// *  instead. The SOKOL_<BACKEND> macro is set by the Revenant CMake      *
// *  target so every TU that pulls in <sokol_gfx.h> sees the same        *
// *  backend selection.                                                  *
// *************************************************************************

#define SOKOL_IMPL

#import <sokol_gfx.h>
#import <sokol_app.h>
#import <sokol_glue.h>
#import <sokol_time.h>
#import <sokol_args.h>

File: src/sokol_impl.c
// *************************************************************************
// *                      RevenantRevisited (port)                         *
// *  sokol_impl.c - Sokol implementation TU for Windows / Linux builds.   *
// *                                                                       *
// *  Non-Apple path: sokol_app + sokol_gfx compile as plain C against     *
// *  either SOKOL_D3D11 (Windows) or SOKOL_GLCORE33 (Linux). Apple ships  *
// *  with sokol_impl.m instead, where the Metal backend requires an       *
// *  Objective-C TU. The SOKOL_<BACKEND> macro is set by the Revenant     *
// *  CMake target so every TU that pulls in <sokol_gfx.h> sees the same   *
// *  backend selection.                                                   *
// *************************************************************************

#define SOKOL_IMPL

#include <sokol_gfx.h>
#include <sokol_app.h>
#include <sokol_glue.h>
#include <sokol_time.h>
#include <sokol_args.h>
